source/dealii/dealii_hierarchy_helpers.cc

```diff
@@ -35,25 +35,22 @@ std::shared_ptr<Operator<VectorType>>
 DealIIHierarchyHelpers<dim, VectorType>::get_global_operator(
     std::shared_ptr<MeshEvaluator> mesh_evaluator)
 {
-  if (_global_operator == nullptr)
```

Comment:
How did this code create a memory leak?

For the new implementation `get_global_operator` sounds a little like a misnomer. I would have expected the former behavior for something called `get` (and probably also `const`). The new one rather creates the operator, doesn't it?

Reply:
So something like https://godbolt.org/z/Vhnizg.

Reply:
yeah something like that

Reply:
I don't have a strong opinion on the name but technically we are not building the global operator because it should already exists in the user's code (this is the system the user wants to solver). So we are getting it from the user but I am open to change the name.